NOnion/Crypto/TorStreamCipher.fs

```diff
@@ -0,0 +1,81 @@
+﻿namespace NOnion.Crypto
+
+open System.Security.Cryptography
+open System
+
+type TorStreamCipher (keyBytes: array<byte>, ivOpt: Option<array<byte>>) =
+
+    [<Literal>]
+    let BlockSize = 16
+
+    let mutable counter: array<byte> = Array.zeroCreate BlockSize
+    let mutable counterOut: array<byte> = Array.zeroCreate BlockSize
+    let mutable keyStreamPointer: int = -1
+
+    let encryptLock: obj = obj ()
+
+    let cipher: RijndaelManaged =
+        new RijndaelManaged (
+            Key = keyBytes,
+            Mode = CipherMode.ECB,
+            Padding = PaddingMode.None
+        )
+
+    do
+        if ivOpt.IsSome && ivOpt.Value.Length = BlockSize then
```

Comment:
using the .Value property of Option is a bad practice (as explained in CONTRIBUTING.md), please use a match instead